File: Codewars_Pascal/CounttheDigit/countthedigit.pas
function NbDig(n, d: Int64): Int64;
var
    i,j,k : Int64;
begin
    j := 0;
    for i := 0 to n do begin
        k := i * i;
    if (d = 0) and (k = 0) then Inc(j);
    while k > 0 do begin
        if (k mod 10) = d then Inc(j);
            k := k div 10;
        end;
    end;
    NbDig := j;
end;

begin
    writeln('Test numbers: '#9, NbDig(5750, 0), ' Answer - > 4700'); 
    writeln('Test numbers: '#9, NbDig(11011, 2), ' Answer - > 9481');
    writeln('Test numbers: '#9, NbDig(12224, 8), ' Answer - > 7733');
end.

File: Codewars_C/7kyu/CounttheDigit/countthedigit.c
#include "stdio.h"

int nbDig(int n, int d) {
    int count = 0;
    for (int i = 0; i <= n; i++) {
        int num = i * i;
        if(num == 0 && d == 0) {
            count++;
        }
        while (num > 0) {
            if (num % 10 == d) {
                count++;
            }
            num /= 10;
        }
    }
    return count;
}

int main() {
    printf("%s", "Test = ");
    printf("%d", nbDig(10, 1));
    printf("%s\n", " answer -> 4");
    printf("%s", "Test = ");
    printf("%d", nbDig(5750, 0));
    printf("%s\n", " answer -> 4700");
    printf("%s", "Test = ");
    printf("%d", nbDig(11011, 2));
    printf("%s\n", " answer -> 9481");
    printf("%s", "Test = ");
    printf("%d", nbDig(12224, 8));
    printf("%s\n", " answer -> 7733");
    printf("%s", "Test = ");
    printf("%d", nbDig(11549, 1));
    printf("%s\n", " answer -> 11905");
    printf("%s", "Test = ");
    printf("%d", nbDig(14550, 7));
    printf("%s\n", " answer -> 8014");
    printf("%s", "Test = ");
    printf("%d", nbDig(8304, 7));
    printf("%s\n", " answer -> 3927");
    printf("%s", "Test = ");
    printf("%d", nbDig(10576, 9));
    printf("%s\n", " answer -> 7860");
    printf("%s", "Test = ");
    printf("%d", nbDig(12526, 1));
    printf("%s\n", " answer -> 13558");
    printf("%s", "Test = ");
    printf("%d", nbDig(7856, 4));
    printf("%s\n", " answer -> 7132");
    printf("%s", "Test = ");
    printf("%d", nbDig(14956, 1));
    printf("%s\n", " answer -> 17267");
    return 0;
}